mne/io/base.py

```diff
@@ -410,6 +410,45 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         """
         raise NotImplementedError
 
+    def _check_bad_segment(self, start, stop, picks, segment_reject=False):
+        """Function for checking the data segment. If the slice is good,
+        returns the data in desired range. If rejected based on
+        annotation, returns description of the bad segment.
+
+        Parameters
+        ----------
+        start : int
+            First sample of the slice.
+        stop : int
+            End of the slice.
+        picks : array of int
+            Channel picks.
+        segment_reject : bool
+            Whether to perform rejection based on annotations.
+            False by default.
+
+        Returns
+        -------
+        data : array | str
+            Data in the desired range (good segment) or description of the bad
+            segment.
+        """
+        if start < 0:
+            return None
+        if segment_reject and self.annotations is not None:
+            annot = self.annotations
+            sfreq = self.info['sfreq']
+            for onset_idx, onset in enumerate(annot.onset):
```

Comment:
I could see this potentially having bad performance. Thinking about making 1000 epochs from data with 1000 annotations and you will be in trouble. The logic should be vectorizable, can you try?